### DISCUSSION TITLE
Features suggestions

### DISCUSSION THREAD

**Discussion by Didayolo** [codalab/codabench#1334] [Ideas]
This discussion is for proposing features for future developments.

# UI Features

### Dark and light  UI (#915)
A feature to toggle light and dark mode for the website. Can also be toggled using system settings 

### Make UI responsive (#719)
Make changes in the UI to make it mobile-friendly.

### Email templates (#1230) 
Design good looking email templates for all the emails sent to participants and organizers

### Theme matching Dialog (#911)
Design theme matching dialog modals

### Expandable text in home page (#1268)

Make the text (history, etc.) expandable only showing few lines first.

### Competition cover (#1854)

Adding a cover photo for competitions. This way we can change the way we are showing the competitions on home page and public page. Instead we can use the competition logo and cover to show nice cards.

***

# Functional Features

### Leaderboard form: a submission rule per challenge phase (#1331)

In "Leaderboard form" we can select, among other options, "Force last" or "Force best".
It would be amazing if you could consider adding a functionality where we can select such option for each challenge phase, such that we could select, for example, to use "Force last" in a development phase, and "Force best" in a test (final) phase.

### Custom Docker Image for submission (#1149)

Allow participants to select the Docker image of their choice.

### Multiple leaderboards (#873)
Possibility of splitting the results across multiple tables can be nice

### Assets directory (#992)
Having a directory in the competition bundle containing files (images, etc.) to which the documentation pages (markdown or HTML) can link to. Right now we need to link images using external URLs instead of having them directly inside the bundle.

### Meta Info in competition page (#1050)
Include meta-data e.g. image, title, description etc. in the competition page

### Notification Center (#1242)
develop a dedicated notification center 

### Submit independently to tasks (#1711)

Right now, submitting to only one task results in having n/a scores to the other tasks. It would be interesting to be able to submit seperately to different tasks and have the results merged on the leaderboard.

### Organizations (#1217, #1240)
Make organizations clear with more functionalities e.g. sharing tasks, datasets etc and more

### Uploading of scoring/ingestion program robust to sub-directories (#968)

Only zipping without directory structure works right now. PR #1143 proposed to make this more robust.

### Send email invitations (#1441)

Invite people to your competition through emails

### Task validation (#1078)

Validate tasks that have successfully run. Unvalidate tasks when someone update them. That needs discussion (see #1078).

***

# Security Features

### Encrypt data (#903)

***

# Documentation

### Dataset submission bundle (#1013)

Add a good "dataset submissions" bundle in competition-examples

**Comment by hzxie** [comment]
[Semantic UI](https://semantic-ui.com/) natively support [responsive design](https://developer.mozilla.org/en-US/docs/Learn_web_development/Core/CSS_layout/Responsive_Design) for mobile devices.
Please follow the [documentation of container](https://semantic-ui.com/elements/container.html) to correct arrange HTML elements.

Instead of using this STUPID solution.

> While forcing the desktop version of your website on mobile devices is generally not recommended due to user experience and accessibility concerns, you can achieve this using a combination of JavaScript and CSS.
> 
> 
> 
> Here's a simple way to force the desktop version for mobile visitors:
> 
> 
> 
> 1. Add a meta tag in the <head> section of your HTML file to set the default viewport scale:
> 
> 
> 
> ```html
> 
> <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
> 
> ```
> 
> 
> 
> 2. Use JavaScript to detect if the visitor is using a mobile device and then modify the viewport meta tag accordingly:
> 
> 
> 
> ```html
> 
> <script>
> 
>   function isMobile() {
> 
>     return /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent);
> 
>   }
> 
> 
> 
>   if (isMobile()) {
> 
>     document.querySelector('meta[name="viewport"]').setAttribute('content', 'width=1024');
> 
>   }
> 
> </script>
> 
> ```
> 
> 
> 
> This script will check if the user is on a mobile device, and if so, it will set the viewport width to a fixed value (e.g., 1024px) to mimic the desktop version. You can adjust the fixed width value according to your website's design.
> 
> 
> 
> However, I would strongly recommend adopting a responsive design approach instead of forcing the desktop version on mobile devices. This will ensure that your website looks and functions well on all devices, providing a better user experience. You can achieve this by using CSS media queries, responsive grids, and flexible layouts. 

 _Originally posted by @Didayolo in [#719](https://github.com/codalab/codabench/issues/719#issuecomment-1540467354)_

**Reply by Didayolo** [reply]
The "solution" you are quoting is not used and was a simple suggestion in a comment.
Codabench is a free, open-source, community-driven project.
You are welcome to make a pull request to contribute to the UI.
Thank you for your comprehension.

**Reply by hzxie** [reply]
I really want to contribute to this project. Ten years ago, I contributed to many similar projects. But after doing a PhD, I really feel like I want to but just don’t have the energy.
